tests/test_ftp.py

```diff
@@ -239,3 +244,49 @@ def test_close(self, mock_ftp, mock_storage):
         file_.is_dirty = True
         file_.read()
         file_.close()
+
+
+class FTPTLSTest(TestCase):
+    def setUp(self):
+        self.storage = ftp.FTPStorage(location=URL_TLS)
+
+    def test_init_no_location(self):
+        with self.assertRaises(ImproperlyConfigured):
+            ftp.FTPStorage()
+
+    @patch("storages.backends.ftp.setting", return_value=URL_TLS)
+    def test_init_location_from_setting(self, mock_setting):
+        storage = ftp.FTPStorage()
+        self.assertTrue(mock_setting.called)
+        self.assertEqual(storage.location, URL_TLS)
+
+    def test_decode_location(self):
+        config = self.storage._decode_location(URL_TLS)
```

Comment:
Given that `_decode_location` is called on `__init__` this test can be changed to

```python
def test_decode_location(self):
    wanted_config = {
        "passwd": "b@r",
        "host": "localhost",
        "user": "foo",
        "active": False,
        "path": "/",
        "port": 2121,
        "secure": True
    }
    self.assertEqual(self.storage._config, wanted_config)
```

Reply:
Check below